pkgs/games/am2rlauncher/default.nix

```diff
@@ -0,0 +1,113 @@
+{ lib,
+  buildDotnetModule,
+  pkgsi686Linux,
+  glibc,
+  gtk3,
+  libappindicator,
+  webkitgtk,
+  e2fsprogs,
+  libnotify,
+  libgit2,
+  openssl,
+  xdelta,
+  file,
+  busybox,
+  openjdk,
+  steam-run,
+  patchelf,
+  fetchFromGitHub,
+  makeDesktopItem }:
+
+buildDotnetModule rec {
+  pname = "AM2RLauncher";
+  version = "2.3.0";
```

Comment:
Should be `unstable-2023-09-03` if you are using a non-tagged commit.

Reply:
Unfortunately, it looks like NuGet doesn't play nicely with non-semver values of `version`. When I try `version = "unstable-2023-10-12`, I get the following build error:

```
error: builder for '/nix/store/qy41fia5d91nn3qfwa58y2hn005z3m83-am2r-launcher-unstable-2023-10-12.drv' failed with exit code 1;
       last 10 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/hdgissird7hpfn8860dqkvazni29ykvm-source
       > source root is source
       > patching sources
       > applying patch /nix/store/66i1s7gz5zw0mpvndid6xr1l3bk8738l-steam-run-binary.patch
       > patching file AM2RLauncher/AM2RLauncherLib/Profile.cs
       > configuring
       > Executing dotnetConfigureHook
       >   Determining projects to restore...
       > /nix/store/dn7x9k9gy7xrnl92phh66i8j3yh0vj5s-dotnet-sdk-6.0.408/sdk/6.0.408/NuGet.targets(132,5): error : 'unstable-2023-10-12' is not a valid version string. (Parameter 'value') [/build/source/AM2RLauncher/AM2RLauncher.Gtk/AM2RLauncher.Gtk.csproj]
       For full logs, run 'nix log /nix/store/qy41fia5d91nn3qfwa58y2hn005z3m83-am2r-launcher-unstable-2023-10-12.drv'.
```

I can contact the maintainers and see if they are willing to do a minor version bump for this package, however.

Reply:
version bump is gonna prove a little difficult

Is `2.3.0-unstable-2023-09-03` acceptable?

Reply:
`2.3.0-unstable-2023-09-03` does allow the package to build and run unlike `unstable-2023-09-03`, but I think it's unconventional for Nix packages. I'll defer to @SuperSamus on that one, since they have more packaging experience than myself.

Reply:
> `2.3.0-unstable-2023-09-03` does allow the package to build and run

Looking [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming), this seems to be allowed.

> Example: Given a project had its latest releases `2.2` in November 2021, and `3.0` in January 2022, a commit authored on March 15, 2022 for an upcoming bugfix release `2.2.1` would have `version = "2.2-unstable-2022-03-15"`.